### PR TITLE
Add screen home - Part 2

### PR DESCRIPTION
**Pull request by Vcky4**
This pull request adds a new screen home and a component VideoFeedItem to the application, enhancing the user experience and providing a valuable feature. 
The screen home has a vertical swipeable list of videos(VideoFeedItem) and layout components that are intuitive and visually appealing. I have thoroughly reviewed the code changes, and everything appears to be in order. 

The following has not been done yet
1. manually pause the video on tap

Please review and merge at your convenience. Thank you!

### CODE REVIEW

File: app/Home.tsx [hunk 1]
```diff
@@ -1,94 +1,123 @@
...
+
+
 export default function ScreenHome() {
   const { primary: colorPrimary } = useColors();
+  const items = [
+    { video: require("@assets/videos/dog.mp4") },
+    { video: require("@assets/videos/girl.mp4") },
+    { video: require("@assets/videos/ballon.mp4") },
+    { video: require("@assets/videos/home-temp.mp4") },
+  ]
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply app/Home.tsx:32]: I've fix this in the next PR

File: app/Home.tsx [hunk 2]
```diff
@@ -1,94 +1,123 @@
...
+import { useRef, useState } from "react";
 
 const DEFAULT_CAMERA_ACTION_COLOR = "#FFF";
 
+//get device width and height
+const { width, height } = Dimensions.get("window");
+
 /**
- * Home screen component
+ * TODO - Build Home screen component
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply app/Home.tsx:17]: Fixed too

File: app/Home.tsx [hunk 3]
```diff
@@ -1,94 +1,123 @@
...
+
 
   return (
     <>
-      <Video
-        source={require("@assets/videos/home-temp.mp4")}
-        shouldPlay
-        isLooping
-        resizeMode={ResizeMode.COVER}
+      <FlatList
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply app/Home.tsx:47]: We can't use ScrollView as users will post thousands of videos and we can't go back to change it to Flatlist then

File: app/Home.tsx [hunk 4]
```diff
@@ -1,94 +1,123 @@
...
         style={[StyleSheet.absoluteFill]}
+        // contentContainerStyle={{
+        //   width: width,
+        //   height: height,
+        // }}
+        data={items}
+        renderItem={({ item, index }) =>
+          <VideoFeedItem item={item} index={index} visibleIndex={visibleIndex} />
+        }
+        keyExtractor={(item, index) => index.toString()}
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply app/Home.tsx:57]: There's no real data for now hence me using an index

When we have real data I'll change it to Id

If you observed both index and item is there which makes it easier to switch

File: src/components/VideoFeedItem.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        resizeMode={ResizeMode.COVER}
+        style={{ height: '100%', width: width, position: "absolute" }}
+
+      // style={[StyleSheet.absoluteFill]}
+      />
+      <View style={{
+        flex: 1,
+        marginHorizontal: 10,
+        marginBottom: "20%",
+        justifyContent: "flex-end",
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply src/components/VideoFeedItem.tsx:49]: done

File: src/components/VideoFeedItem.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+          }}>
+            <Image style={styles.profileImg} source={profileImg} />
+            <Text variant="titleMedium" style={[styles.headActionText, { width: '75%', }]}>
+              @glory.jgy {'\n'}
+              <Text variant="bodyLarge" style={{ color: DEFAULT_CAMERA_ACTION_COLOR }}>
+                My very first podcast, it was really fun and I learnt so much just in one day.
+              </Text>
+            </Text>
+          </View>
+          <TouchableOpacity style={{
```
[the remark this answers is not in the document]
Comment (Vcky4) [reply src/components/VideoFeedItem.tsx:88]: I changed my mind

**Comment by Vcky4** [comment]
Video pause and play has been done

File: assets/videos/girl.mp4
Comment (firasrg) [inline assets/videos/girl.mp4:1]: Can you please compress this video to reduce it's size to something inferior than 7.56 MB?

File: src/components/VideoFeedItem.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+const DEFAULT_CAMERA_ACTION_COLOR = "#FFF";
+
+//get device width and height
+const { width, height } = Dimensions.get("window");
+
+/**
+ * TODO - Build Home screen component
+ *
+ * @constructor
+ */
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:19]: This is a wrong comment!

File: src/components/VideoFeedItem.tsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+  const [shouldPlay, setShouldPlay] = useState(false)
+
+  //set play state
+  useEffect(() => {
+    setShouldPlay(index === visibleIndex)
+  }, [visibleIndex])
+
+  const togglePlay = () => {
+    setShouldPlay(!shouldPlay)
+  }
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:43]: I think that, **it's recommended** to follow the Imprative playback APIs when controlling videos state.  
Please follow [the guide's example](https://docs.expo.dev/versions/latest/sdk/video/#usage). 

There are some **useful methods** in Video components that can be used to play/pause.

File: src/components/VideoFeedItem.tsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+  return (
+    <TouchableWithoutFeedback
+      onPress={togglePlay}
+      style={{ flex: 1 }}>
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:48]: Why I still see style declarations thrown into component?

File: src/components/VideoFeedItem.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+            <View style={{
+              justifyContent: "center",
+              alignItems: "center",
+            }}>
+              {
+                icons.map((icon, index) =>
+                  <TouchableOpacity key={index}
+                    style={{
+                      paddingVertical: 12,
+                    }}>
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:68]: Same note for styles.

File: src/components/VideoFeedItem.tsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+          isLooping
+          resizeMode={ResizeMode.COVER}
+          style={[StyleSheet.absoluteFill]}
+        />
+        <View style={styles.bottomContainer}>
+          <View style={styles.rightActionsContainer}>
+            <View style={{
+              justifyContent: "center",
+              alignItems: "center",
+            }}>
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:62]: Same note for styles, and please consider using our **variants**.

File: src/components/VideoFeedItem.tsx [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+              }
+
+            </View>
+          </View>
+
+          <View
+            style={{
+              flexDirection: "row",
+              justifyContent: "space-between",
+              marginTop: 14,
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:81]: Same note for styles.

File: src/components/VideoFeedItem.tsx [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+            }}
+          >
+            <View style={{
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:84]: Same note for styles

File: src/components/VideoFeedItem.tsx [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+            }}>
+              <Image style={styles.profileImg} source={profileImg} />
+              <Text variant="titleMedium" style={[styles.headActionText, { width: '75%', }]}>
+                @glory.jgy {'\n'}
+                <Text variant="bodyLarge" style={{ color: DEFAULT_CAMERA_ACTION_COLOR }}>
+                  My very first podcast, it was really fun and I learnt so much just in one day.
+                </Text>
+              </Text>
+            </View>
+            <TouchableOpacity >
```
Comment (firasrg) [inline src/components/VideoFeedItem.tsx:95]: Please add `<TouchableOpacity >` inside `ButtonAddSimple` component, as it's button, so It has to be touchable.

**Review by firasrg — CHANGES_REQUESTED** [review]

**Comment by firasrg** [comment]
Recap Review Feedback: Code Organization, Commit History, and Documentation

@Vcky4 I've completed my review of the pull request and would like to provide some general feedback regarding the code organization, commit history, and adherence to the README and CONTRIBUTING guide.

### Code Organization:
It appears that the code could benefit from better organization and structure. For example, the styles. I'd like to see the use of variants. The styles must be declared separately, unless there is a reason to declare them into the components. Modularity makes it easier for other developers to understand the codebase.

### Commit History:
I noticed that there are a significant number of commits in this pull request (13 in total) and I assume more after my recent review. While it's not inherently bad to have multiple commits, it would be beneficial to group related changes together. This helps provide a clearer and more concise overview of the changes made. Consider **squashing** some commits that address similar functionality or bug fixes.

### README and CONTRIBUTING Guide:
Please make sure to follow the guidelines outlined in the README and CONTRIBUTING guide. These documents serve as **essential** references for developers. Adhering to the guidelines ensures consistency across the codebase and helps streamline the development process. Take some time to review and update your code accordingly.

Remember, we strive for a well-structured and maintainable codebase, so taking these considerations into account will contribute to the overall quality of the project.

Feel free to reach out if you have any questions or require further clarification on any of the points mentioned above.

Best regards,

**Pull request merged**